doc/case-studies/timelock/claim.tla

```diff
@@ -0,0 +1,108 @@
+(*
+ * Claim method monitors for the Timelock contract
+ *
+ * Andrey Kuprianov, 2024
+ *)
+---- MODULE claim ----
+
+EXTENDS timelock
+
+(*
+    @typeAlias: claimArgs = { 
+        claimant: $address
+    };
+*)
+claim_typedefs == TRUE
+
+
+\* @type: ($claimArgs) => Bool;
+MustFail_claim_Unauthorized(args) == 
+    ~authorized(args.claimant)
+
+MustFail_claim_NoBalanceRecord(args) == 
+    ~instance_has("Balance")
+
+\* @type: ($claimArgs) => Bool;
+MustFail_claim_NotClaimant(args) == 
+    \A i \in DOMAIN Balance.claimants: 
+        Balance.claimants[i] /= args.claimant
+
+\* One success condition: correctly claimed before time bound
+MustPass_claim_BeforeTimeBound(args) ==
+    /\ Balance.time_bound.kind = "Before" 
+    /\ env.ledger_timestamp <= Balance.time_bound.timestamp
```

Comment:
None of this is blocking, but from how the source code is structured:
https://github.com/yuzurush/soroban-contracts-101/blob/4f64753310b26c7853de1c9efcc0f6f2b349c15c/timelock/src/lib.rs#L104-L106
I see this (and the one below) conceptually as a `MustFail_...` predicate, since it panics if the condition doesn't hold, and executes a bunch of code afterwards if it does. So in my view:
```suggestion
MustFail_claim_BeforeTimeBound(args) ==
    /\ Balance.time_bound.kind = "Before" 
    /\ env.ledger_timestamp > Balance.time_bound.timestamp
```